src/graph.py

```diff
@@ -0,0 +1,122 @@
+"""A gragh data structure."""
+
+
+class Graph(object):
+    """Graph."""
+
+    def __init__(self):
+        """Create the dictionary that will be the graph."""
+        self.g = {}
+
+    def nodes(self):
+        """Return a list of all the nodes in the graph."""
+        nodes = []
+        for key in self.g:
+            nodes += [key]
+        return nodes
+
+    def edges(self):
+        """Return a list of all the edges in the graph."""
+        edges = []
+        for key in self.g:
+            for edge in self.g[key]:
+                edges += [(key, edge)]
+        return edges
+
+    def add_node(self, n):
+        """Add a node to the graph."""
+        self.g.setdefault(n, [])
+
+    def add_edge(self, n1, n2):
+        """Add an edge between two nodes."""
+        weight = 0
+        if type(n1) is int or type(n1) is float:
+            weight += n1
+        else:
+            weight += 1
+        if type(n2) is int or type(n2) is float:
+            weight += n2
+        else:
+            weight += 1
+        try:
+            self.g[n1] += [(n2, weight)]
+        except KeyError:
+                self.g[n1] = [(n2, weight)]
+
+    def del_node(self, n):
+        """Delete a node from the graph."""
+        try:
+            del self.g[n]
+        except KeyError:
+            raise KeyError('No node exists.')
+
+    def del_edge(self, n1, n2):
+        """Delete the edge between two nodes."""
+        try:
+            for i in self.g[n1]:
+                if i[0] is n2:
+                    self.g[n1].remove(i)
+                    break
+        except ValueError:
+            raise ValueError('Edge does not exist.')
+
+    def has_node(self, n):
+        """Find a specific node in the graph."""
+        return n in self.g
+
+    def neighbors(self, n):
+        """Return all the edges for n."""
+        try:
+            nbrs = []
+            for i in self.g[n]:
+                nbrs += [i[0]]
+            return nbrs
+        except KeyError:
+            raise KeyError('No node exists.')
```

Comment:
The message implies that no node exists in the graph. Only the provided node doesn't exist.